prover/crates/bin/witness_generator/src/leaf_aggregation.rs

```diff
@@ -155,7 +152,10 @@ impl JobProcessor for LeafAggregationWitnessGenerator {
         job: LeafAggregationWitnessGeneratorJob,
         started_at: Instant,
     ) -> tokio::task::JoinHandle<anyhow::Result<LeafAggregationArtifacts>> {
-        tokio::task::spawn_blocking(move || Ok(Self::process_job_sync(job, started_at)))
+        let object_store = self.object_store.clone();
+        tokio::task::spawn(async move {
+            Ok(Self::process_job_impl(job, started_at, object_store).await)
+        })
```

Comment:
This is dangerous. Performing blocking work in the tokio runtime threads can cause the reactor to hand and/or behave unexpectedly otherwise.
If you need to mix blocking and non-blocking work, the preferred way is to obtain a runtime handle via `tokio::runtime::Handle::current()` and pass it into the `spawn_blocking` context, so that you can `handle.block_on(async_stuff)` inside.

Reply:
Indeed, I missed it somehow.

Reply:
And what about this place?

https://github.com/matter-labs/zksync-era/blob/8b6add2d654235e60ad1b2b5e9cddbb695dcf4d3/prover/crates/bin/witness_generator/src/basic_circuits.rs#L508-L527

Reply:
This code isn't blocking, yes